.github/workflows/test-package-lib.yml

```diff
@@ -0,0 +1,32 @@
+name: Package Library
+on:
+  pull_request:
+  push:
+    branches-ignore:
+      - 'gh-pages'
+
+jobs:
+  test-package-lib:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+          - node: '14'
+            engine-strict: 'false'
+          - node: '16'
+            engine-strict: 'true'
+          - node: '18'
+            engine-strict: 'false'
```

Comment:
While I was working on this I took the opportunity to drop node 12 testing (which is now out of LTS), add node 18 and update the package changelog and package.json